plugin/concurrency.go

```diff
@@ -0,0 +1,76 @@
+package plugin
+
+import (
+	"log"
+	"sync"
+)
+
+const defaultMaxCalls = 50
+const defaultMaxPerCall = 10
+
+type ConcurrencyManager struct {
+	mut        sync.Mutex
+	totalCalls int
+	// map of max calls allowed for each func
+	callMaxMap map[string]int
+	// map of current calls for each func
+	callMap map[string]int
+}
+
+func newConcurrencyManager() *ConcurrencyManager {
```

Comment:
for now remove (comment out) `defaultMaxPerCall` - our config schema does not include it

We need to move defaultMaxCalls into the concurrency manager and populate from the plugin  `DefaultHydrateConfig.MaxConcurrency` (add same var to `ConcurrencyManmager` )

callMaxMap either needs to be populated when we start concurrency manager from the hydrate configs of the required hydrate functions, or removed and use the hydrate config directly